src/Disks/DiskRemote.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <Client/RemoteFSConnectionPool.h>
+
+#include <Disks/DiskLocalCheckThread.h>
+#include <Disks/IDisk.h>
+#include <IO/ReadBufferFromFile.h>
+#include <IO/ReadBufferFromFileBase.h>
+#include <IO/WriteBufferFromFile.h>
+#include <Poco/Util/AbstractConfiguration.h>
+#include <Common/logger_useful.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int NOT_IMPLEMENTED;
+}
+
+class DiskRemote : public IDisk
```

Comment:
This name is a bit misleading, because it intersects with our existing remote disks: s3, hdfs, azure. In fact some time ago we already has a class `DiskRemote` which was used for s3 and other, but recently it was moved into another abstraction. So may be name this disk as `DiskRemoteFSOverClickHouseServer`? same in `DiskType.h`